### PR TITLE
Added PAPERS domain

### PR DESCRIPTION
**Pull request by johnSamilin**

### CODE REVIEW

File: src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+function Highlight(props) {
+  const { field, value } = props;
+
+  return (    
+    <div {...classes()}>
+      <span {...classes('field')}>{field}</span>
+      <div {...classes('value')}>
+        {value.split(detectorRegexp).map(part => <Part value={part} />)}
```
Comment (pavgra) [inline src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx]: Quite expensive to do split each time. Let's move to selectors

File: src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  * Created: Monday, March 19, 2018 11:24 AM
+  *
+  */
+
+import React from 'react';
+import BEMHelper from 'services/BemHelper';
+
+import './style.scss';
+
+const detectorRegexp = /(\[b\][^\[]*\[\/b\])/g;
```
Comment (pavgra) [inline src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx]: Why two regexps?

File: src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+const isHighlightedRegexp = /^(\[b\])[^\[]*(\[\/b\])$/g;
+const classes = new BEMHelper('highlight');
+
+function Part({ value }) {
+  const isMatch = isHighlightedRegexp.test(value);
+  return (<span {...classes('part', { matched: isMatch })}>
+    {value.replace(/\[\/?b\]/g, ' ')}
```
Comment (pavgra) [inline src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx]: Move it to variable not to duplicate

File: src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx (hunk 1 above)
Comment (johnSamilin) [reply src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx]: they will do the same job each time, won't they?

File: src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx (hunk 2 above)
Comment (johnSamilin) [reply src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx]: cause one seeks [b] in the start of the string, and [/b] in the end

File: src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx (hunk 3 above)
Comment (johnSamilin) [reply src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx]: this is not a duplication, there are 3 different regexps

File: src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx (hunk 2 above)
Comment (pavgra) [reply src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx]: Why do job twice? Use match

File: src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx (hunk 1 above)
Comment (pavgra) [reply src/modules/Portal/components/Search/Results/List/ResultItem/components/Highlight/presenter.jsx]: Then why memoizers were created?

**Pull request merged**